NitroxPatcher/Patches/Dynamic/WaterParkCreature_BornAsync_Patch.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Prevents fishing from being born in a WaterPark if the said WaterPark is not simulated by the local player.
+/// Syncs fish birth.
+/// </summary>
+public sealed partial class WaterParkCreature_BornAsync_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = AccessTools.EnumeratorMoveNext(Reflect.Method(() => WaterParkCreature.BornAsync(default, default, default)));
+
+    /*
+     * MODIFIED:
+     * if (creaturePrefabReference == null || !creaturePrefabReference.RuntimeKeyIsValid())
+     * BECOMES
+     * if (creaturePrefabReference == null || !creaturePrefabReference.RuntimeKeyIsValid() || !WaterParkCreature_BornAsync_Patch.CanBeBorn(waterPark))
+     * 
+     * INSERTED:
+     * result.SetActive(true);
+     * waterPark.AddItem(pickupable);
+     * WaterParkCreature_BornAsync_Patch.Callback(pickupable); <--- INSERTED LINE
+     */
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        // Finds the waterPark parameter reference so that we can duplicate it
+        CodeMatcher waterParkMatcher = new CodeMatcher(instructions).MatchEndForward([
```

Comment:
Would add a small bit of info to why we need to do it this way (IEnumerator shenanigans)